render/hyperscript.js

```diff
@@ -27,23 +27,15 @@ function compileSelector(selector) {
 function execSelector(state, attrs, children) {
 	var hasAttrs = false, childList, text
 	var className = attrs.className || attrs.class
+	var classAttr = hasOwn.call(attrs, "class") ? "class" : "className"
 
 	for (var key in state.attrs) {
-		if (hasOwn.call(state.attrs, key)) {
+		if (hasOwn.call(state.attrs, key) && key !== "className") {
 			attrs[key] = state.attrs[key]
 		}
 	}
 
-	if (className != null) {
-		if (attrs.class != null) {
-			attrs.class = undefined
-			attrs.className = className
-		}
-
-		if (state.attrs.className != null) {
-			attrs.className = state.attrs.className + " " + className
-		}
-	}
+	if (className != null || state.attrs.className != null) attrs[classAttr] = state.attrs.className == null ? className || "" : state.attrs.className + (className ? " " + className : "")
```

Comment:
Could you split this apart to remove some redundant branching and make the logic clearer?

```js
if (state.attrs.className != null) {
  attrs[classAttr] = state.attrs.className + (className ? " " + className : "")
} else if (className != null) {
  attrs[classAttr] = className || ""
}
```